app/services/include_params_validator.rb

```diff
@@ -22,6 +22,26 @@ def fully_validate!
     raise ValidationError, self if invalid?
   end
 
+  # Active Model Serializers return all members of a nested chain
+  #
+  # For example:
+  #   'foo.bar.baz' is equivalent to ['foo', 'foo.bar', 'foo.bar.baz']
+  #
+  # To reflect this behaviour in the validation we need to support all variations
+  # of nested chains by preformatting them in this validator.
+  def formatted_supported_relationships
+    formatted_supported_relationships = Set.new
+
+    supported_relationships.each do |chain|
+      split_chain = chain.split('.')
+      split_chain.each_with_index do |_, index|
+        formatted_supported_relationships << split_chain[0..index].join('.')
```

Comment:
have you checked that this line does create duplication?
for example, if ['foo', 'foo.bar', 'foo.bar.baz'] is passed, we do not want any repetion of foo and foo.bar

Reply:
Using Set.new :)